src/common/snippets/include/snippets/lowered/port_descriptor.hpp

```diff
@@ -36,15 +36,18 @@ class PortDescriptor {
     explicit PortDescriptor(const ov::Output<const ov::Node>& node,
                             VectorDims subtensor_shape = {},
                             std::vector<size_t> layout = {});
-    PortDescriptor(VectorDims shape, VectorDims subtensor_shape, std::vector<size_t> layout = {});
-    PortDescriptor() = default;
+    PortDescriptor(VectorDims shape, VectorDims subtensor_shape, std::vector<size_t> layout = {}, Reg reg = {});
+    PortDescriptor(VectorDimsPtr shape, VectorDims subtensor_shape, std::vector<size_t> layout = {}, Reg reg = {});
+    PortDescriptor();
 
-    const VectorDims& get_shape() const {return m_tensor_shape;}
+    const VectorDims& get_shape() const;
+    const VectorDimsPtr& get_shape_ptr() const;
```

Comment:
Two similar method look confusing.
Can we remove `get_shape()` then? Because if we need a shape by value we can just get pointer and dereference it.
The same is true for `set_shape()`

Reply:
Discussed offline: decided to make `get_shape_ptr()` and `set_shape_ptr()` private, since they are needed only for a couple of classes

Reply:
@IvanNovoselov applied in the commit https://github.com/openvinotoolkit/openvino/pull/24115/commits/0277e0bf5eb1e28c0cb2aaf64a20ec75e79ec6ba. Could you take a look one more time please?